app/models/work_version.rb

```diff
@@ -314,8 +314,31 @@ def form_partial
            :auto_generated_thumbnail?,
            :thumbnail_url, to: :work
 
+  def citation_display
+    return unless Work::Types.data_and_code.include?(work_type)
+
+    "#{creators_citation_display}(#{year_published}). #{title} [Data set]. Scholarsphere.#{doi_url}"
+  end
+
   private
 
+    def creators_citation_display
+      formatted_creators = ''
+      creators.each_with_index do |creator, index|
+        formatted_creators += "#{creator.surname}, #{creator.given_name}"
+        formatted_creators += index == creators.length - 1 ? ' ' : '; '
+      end
+      formatted_creators
+    end
+
+    def doi_url
+      doi.blank? ? '' : " https://doi.org/#{doi}"
+    end
+
+    def year_published
+      Date.edtf(published_date).try(:year)
+    end
```

Comment:
Can this stuff be removed since it is in the ViewComponent?

Reply:
Yes it can. Thanks for catching that!